Component for replacing the React Mosaic Library

AbhinavThukral97 opened the pull request.

houshuang committed:
Fixing number formatting in template (#1910)

AbhinavThukral97 committed:
Split Window Component

AbhinavThukral97 committed:
Linting

dependabot[bot] committed:
Bump eslint-utils from 1.4.0 to 1.4.3 in /frogapi-test (#1909)

Bumps [eslint-utils](https://github.com/mysticatea/eslint-utils) from 1.4.0 to 1.4.3.
- [Release notes](https://github.com/mysticatea/eslint-utils/releases)
- [Commits](https://github.com/mysticatea/eslint-utils/compare/v1.4.0...v1.4.3)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump sshpk from 1.13.1 to 1.16.1 in /frogapi-test (#1911)

AbhinavThukral97 committed:
Quick UI Fixes (#1913)

houshuang commented:
@lfaucon Can you have a look at this? The idea is to replace React Mosaic with something very simple, since it's currently increasing our bundle size with .5 MB, and has a lot of functionality (like resizing) which we never use... 

I think we still want some kind of simple title bar, similar to what we have now. Also not sure about the odd-number layouts, like 3 - should we have 3 side by side or one taking half, and the two others taking two quarters like now (downside is that we currently can't choose which goes where). 

![Screenshot 2019-11-15 at 08 35 52](https://user-images.githubusercontent.com/61575/68925370-2ae54480-0783-11ea-8341-863ab8d7062a.png)

lfaucon commented:
@AbhinavThukral97 This looks great! Good work

For the view with 3 components, I would prefer one big (half of the screen) and two small (quarter of the screen). Similarly to what we have now already.

![Screenshot from 2019-11-15 09-49-31](https://user-images.githubusercontent.com/8818081/68929914-92a08d00-078d-11ea-83bb-2996ddc1f971.png)

Also, when we replace ReactMosaic, it would be good to have the same topbar on all the children (Like on the screenshot: "Chat (individual/Chen Li)")

@houshuang Are we sure that the students never used the resizing?

Future improvements could be:
- Children can be toggled into full screen
- Children can be minimized (For example if we have 4 windows and 2 are minimized, then the 2 left should fit as a two-children layout)

houshuang commented:
Rather than focusing on resizing and minimizing, I'd rather we move towards
what I've been hoping for for a very long time, which is to stop treating
all activity types the same, but instead special case instructions, social
presence and chat, etc.

lfaucon committed:
Fix issue with removing template (#1916)

AbhinavThukral97 committed:
Editing 3 window view

AbhinavThukral97 committed:
Adding titles to windows

AbhinavThukral97 commented:
I've changed the 3-window view as mentioned and added the functionality for title/topbar using a window component similar to the one currently being used. 
I'll start replacing the Mosaic component with this and test it out quickly.
Let me know if there are any other changes/suggestions :)

houshuang commented:
Looks good to me. Perhaps a different color for the top bar to distinguish? You might also need to add some margin to the body, but you'll see once you try implementing it and rendering real activity types (the preview window is the quickest way of seeing it in action with many different activity types/window configurations).

AbhinavThukral97 committed:
Replacing the component

AbhinavThukral97 committed:
Ratio Fix

houshuang committed:
Removing libraries, not working

houshuang commented:
@AbhinavThukral97 I tried cleaning up a tiny bit, removing the node libraries and the commented code. Currently the Preview page crashes immediately, not quite sure why? If we can get everything to work well and the checks to be green, we can merge.

AbhinavThukral97 committed:
TopBar Design + Preview Fix

AbhinavThukral97 committed:
Merge branch 'ReactMosaic' of https://github.com/chili-epfl/FROG into ReactMosaic

AbhinavThukral97 committed:
Cleanup + Linting

AbhinavThukral97 commented:
I've fixed the Preview section and cleaned the rest of the code. I think we are good to merge now. Let me know if there are any further changes.

houshuang commented:
@lfaucon This looks good to me, can you also review since this touches on very crucial code? The advantage of this will be that we can get rid of a 0.5 MB dependency, and it might also make it easier to customize the interface more in the future.

AbhinavThukral97 committed:
Removing the console print

houshuang committed:
Fixing yarn

AbhinavThukral97 committed:
Tests

AbhinavThukral97 committed:
Flow fix

AbhinavThukral97 committed:
Merge branch 'ReactMosaic' of https://github.com/chili-epfl/FROG into ReactMosaic

lfaucon committed:
Merge branch 'develop' into ReactMosaic

lfaucon commented on frog/imports/client/Preview/Content.jsx:
```diff
@@ -233,51 +251,23 @@ const ContentController = ({
+            ? [
+                ['dashboard', 'dashboard'],
+                ...users.map((name, idx) => [name, instances[idx]])
+              ].map(arr => {
```
To be more easily read. This could be re-written 
`([name, instance ]) => { ... } `

lfaucon replied:
I did it

lfaucon commented on frog/imports/client/Preview/index.js:
```diff
@@ -155,8 +154,4 @@ class PreviewPage extends React.Component<any, any> {
-  <ErrorWrapper>
-    <PreviewPage {...props} />
-  </ErrorWrapper>
-);
```
Why did you remove the `ErrorWrapper`? Does this have something to do with the windowing component?

AbhinavThukral97 replied:
I don't remember removing this. Maybe this was a change merged after I started this branch.

lfaucon commented on frog/imports/client/StudentView/Runner.jsx:
```diff
@@ -163,15 +163,12 @@ const Runner = ({ path, activity, sessionId, object, paused }) => {
-      draggable={false}
+    <ActivityWindow
       key={activity._id}
-      path={path}
```
What was this props `path` that you remove?

AbhinavThukral97 replied:
'path' was a prop for React Mosaic's component that's used to expand split windows.

lfaucon replied:
ok :)

lfaucon committed:
Small readability change

lfaucon committed:
CSS update to add shadow on the windows

lfaucon committed:
Adding back Error Wrapper
